src/core/Global.ttslua

```diff
@@ -825,3 +779,146 @@ function urldecode(str)
     function (h) return string.char(tonumber(h, 16)) end)
   return str
 end
+
+---------------------------------------------------------
+-- Option Panel related functionality
+---------------------------------------------------------
+
+-- loads the default options
+function onClick_defaultSettings()
+  print("Dummy: Load default settings")
+end
+
+-- called by toggling an option
+function onClick_toggleOption(_, id)
+  local state = self.UI.getAttribute(id, "isOn")
+
+  -- flip state (and handle stupid "False" value)
+  if state == "False" then
+    state = true
+  else
+    state = false
+  end
+
+  self.UI.setAttribute(id, "isOn", state)
+  optionPanel[id] = state
+  applyOptionPanelChange(id, state)
+end
+
+-- sets the option panel to the correct state (corresponding to 'optionPanel')
+function updateOptionPanelState()
+  for id, enabled in pairs(optionPanel) do
+    if enabled then
+      self.UI.setAttribute(id, "isOn", true)
+    end
+  end
+end
+
+-- handles the applying of option selections and calls the respective functions based
+---@param id String ID of the option that was selected or deselected
+---@param state Boolean State of the option (true = enabled)
+function applyOptionPanelChange(id, state)
+  -- option: Snap tags
+  if id == "useSnapTags" then
+    playmatAPI.setLimitSnapsByType(state, "All")
+
+  -- option: Draw 1 button
+  elseif id == "showDrawButton" then
+    playmatAPI.showDrawButton(state, "All")
+
+  -- option: Clickable clue counters
+  elseif id == "useClueClickers" then
+    playmatAPI.clickableClues(state, "All")
+
+    -- update master clue counter
+    getObjectFromGUID("4a3aa4").setVar("useClickableCounters", state)
+  
+  -- option: Show token arranger
+  elseif id == "showTokenArranger" then
+    -- delete previously pulled out tokens
+    for _, token in ipairs(getObjectsWithTag("to_be_deleted")) do token.destruct() end
```

Comment:
Don't know the timing on this, but seems like this should happen in onLoad or onSave of the arranger?

Reply:
Good point - moving this to `onDestroy` of the TA